src/mpmc_queue.mli

```diff
@@ -0,0 +1,28 @@
+(** A lock-free multi-producer, multi-consumer, thread-safe queue. 
+    
+    Note: in extreme cases the structure may occassionaly violate FIFO 
+    order. 
+*)
+
+type 'a t = private {
+  array : 'a Option.t Atomic.t Array.t;
+  head : int Atomic.t;
+  tail : int Atomic.t;
+  mask : int;
```

Comment:
I'm asserting on these in the tests. I guess it could be made a little cleaner in the following way: 

```
type 'a t 

... 

module Testing : sig 
  val assert_empty_invariants : unit -> unit 
end 
```